libbeat/docs/shared-autodiscover.asciidoc

```diff
@@ -172,6 +172,25 @@ The `kubernetes` autodiscover provider has the following configuration settings:
 
 include::../../{beatname_lc}/docs/autodiscover-kubernetes-config.asciidoc[]
 
+====== Manually Defining Ports with Kubernetes
```

Comment:
@andrewvc 
Here's the text we talked about:

```
Declare exposed ports in your pod spec if possible. Otherwise, you will need to use multiple templates with complex filtering rules. The {port} variable will not be present, and you will need to hardcode ports. Example: `{data.host}:1234`

When ports are not declared, Autodiscover generates a config using your provided template once per pod, and once per container. These generated configs are deduplicated after they are generated. If the generated configs for multiple containers are identical, they will be merged into one config. 

Pods share an identical host. If only the `{data.host}` variable is interpolated, then one config will be generated per host. The configs will be identical. After they are deduplicated, only one will be used. 
```
Please let me know if you'd like to discuss. Thanks!